doc/double_deposit.md

```diff
@@ -0,0 +1,69 @@
+Similar to ICA, Sending an ibc packet from chain A, include signature of chainB user's signature and execute Tx on chain B according to the signature verification result.
+
+a tx structure will similar to this:
+
+```go
+{
+     senders: []string{"Achain1xxxxxxxxx", "Bchain1xxxxx"},
+     tokens: []*sdk.Coin{{Denom: "aside", Amount: 1000},{Denom: "bside", Amount: 1000}},
+     signature: []byte
+}
+```
+
+steps:
+
+1. construct a tx on chain A
+   - build the Counter party chain sender's signature.
+   ```go
+   depositTx := &types.EncounterPartyDepositTx{
+   			AccountSequence: 1,
+   			Sender:          chainBAddress,
+   			Token:           &sdk.Coin{Denom: chainBDenom, Amount: sdk.NewInt(initialY)},
+   	}
+    rawDepositTx := types.CDC.MustMarshal(depositTx)
+    signedTx := priv.Sign(rawDepositTx)
+   ```
+   - build double deposit message.
+   ```go
+   msg := types.NewMsgDoubleDeposit(
```

Comment:
First of all, not sure `double deposit` is a right name here.  `Crosschain Deposit` or `Two-assets Deposit` would be better.  also want to hear oppinions from @qiu-shane  @EmilGeorgiev  Edward

Reply:
to be clear, I would like to recommend the structure like this:

```js
{
   poolid: "123",
   local_deposit: {
        sender: "",
        token: "100uatom",
   },
   remote_deposit: {
        sequence: ""
        depositor: "",
        token: "200uside",
       signature: ""
    }
}

Reply:
No opinion yet on a proposed name.  Just FYI In Osmosis -- it is called `AddLiquidity`.  They also allow users to add liquidity single-sided, which means to use one token that is automatically swapped 50:50 before adding into the pool.  Bringing this up so we can be strategic with naming.

`Crosschain Deposit` seems interesting.

Reply:
in this step, I can implement that structure, but if we wanna implement balancer white paper, it might consider multi dimensional pool in future

Reply:
> I'm thinking of Multi-asset deposits in the future. At that time, we Will have Multi-dimensions, and users can send various asset lists []sdk. Tokens list which he owned, In fact balancer pool is designed for that. But we only limited dimention to two

Yeah, Balancer support at most 8 assets. how much complex would bring to this module. if not much. we can increase the limit from 2 to 8.  this will increase performance of liquidity